lib/api/src/externals/function.rs

```diff
@@ -911,7 +934,7 @@ impl<T: VMDynamicFunction> VMDynamicFunctionCall<T> for VMDynamicFunctionContext
                 ret.write_value_to(values_vec.add(i));
             }
             Ok(())
-        }));
+        })); // We get an extern ref drop at the end of this block that we don't want
```

Comment:
Oh this comment is about the efficiency of our ref counting, we were getting a drop here (and had to do an inc because of it), this comment is more just about how due to a drop at that point, we're wasting work (incing and then decing, when we could just do neither)

Reply:
I'll revisit it and do something here